blaze/expr/collections.py

```diff
@@ -100,10 +100,32 @@ def sort(child, key=None, ascending=True):
     ascending : bool, optional
         Determines order of the sort
     """
+    if ascending not in (True, False):
+        msg = "ascending must be True or False, given {}"
+        raise ValueError(msg.format(ascending))
     if not isrecord(child.dshape.measure):
-        key = None
+        if key is None or isinstance(key, _strtypes):
+            # Handle this case separately.
+            return Sort(child, None, ascending)
```

Comment:
The code you showed is sort of ambigious with a dshape like: `var * {name: var * {name: A}}`. Do you know if there is any code in the wild that does this with `sort`?

Side note, it looks like `pd.Series.sort` just ignores the first argument (though it is named 'axis'.